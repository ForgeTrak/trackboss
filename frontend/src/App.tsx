import React, { useContext, useEffect } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { UserContext, UserContextProvider } from './contexts/UserContext';
import Dashboard from './pages/Dashboard';
import MemberListPage from './pages/MemberListPage';
import Settings from './pages/Settings';
import CalendarPage from './pages/CalendarPage';
import SignUpPage from './pages/SignUpPage';
import me from './controller/api';
import MemberCommunicationsPage from './pages/MemberCommunicationsPage';
import ApplicationForm from './pages/ApplicationForm';
import RaceAdministration from './pages/RaceAdministration';
import EarlySeasonPage from './pages/EarlySeasonPage';

export function App() {
    const { state, update } = useContext(UserContext);
    const location = useLocation();
    useEffect(() => {
        async function updateState(token: string) {
            const user = await me(token);
            update({ loggedIn: true, token, user, storedUser: undefined });
        }
        if (!state.loggedIn && !location.pathname.includes('apply')) {
            const hash = location.hash.split('#id_token=')[1];
            if (hash) {
                const token = hash.split('&')[0];
                updateState(token);
            } else {
                // this is the only reasonable way to do this other than repeated string concatenation
                // eslint-disable-next-line max-len
                const authTarget = `${process.env.REACT_APP_AUTH_URL}/login?client_id=${process.env.REACT_APP_CLIENT_ID}&response_type=token&scope=email+openid+phone&redirect_uri=${window.location.origin}`;
                window.location.href = authTarget;
            }
        }
    }, [state]);

    return (
        <Routes>
            <Route path="/" element={<Dashboard />} />
            <Route path="calendar" element={<CalendarPage />} />
            <Route path="members" element={<MemberListPage />} />
            <Route path="settings" element={<Settings />} />
            <Route path="calendar/signups/:date/:eventId/:eventType" element={<SignUpPage />} />
            <Route path="jobs" element={<RaceAdministration />} />
            <Route path="early" element={<EarlySeasonPage />} />
            <Route path="communicate" element={<MemberCommunicationsPage />} />
            <Route path="apply" element={<ApplicationForm />} />
        </Routes>
    );
}

export function AppWrapper() {
    return (
        <UserContextProvider>
            <App />
        </UserContextProvider>
    );
}
