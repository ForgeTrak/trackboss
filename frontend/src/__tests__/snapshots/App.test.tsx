import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { AppWrapper } from '../../App';

jest.mock('react-gauge-chart', () => 'GaugeChart');

describe('AppWrapper', () => {
    it('renders correctly', () => {
        const appWrapper = renderer.create(
            <BrowserRouter>
                <AppWrapper />
            </BrowserRouter>,
        ).toJSON();
        expect(appWrapper).toMatchSnapshot();
    });
});
