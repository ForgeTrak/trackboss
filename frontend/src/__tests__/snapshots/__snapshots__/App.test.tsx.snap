// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly parses a key after a redirect 1`] = `[Function]`;

exports[`AppWrapper renders correctly 1`] = `
Array [
  <div
    className="chakra-stack css-p747s7"
  >
    <div>
      <div
        className="css-arm2n8"
      >
        <div
          className="chakra-stack css-84zodg"
        >
          <button
            aria-label="Back"
            className="chakra-button css-76fq9n"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              focusable={false}
              height="lg"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="lg"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
              />
            </svg>
          </button>
          <div>
            <button
              aria-label="Menu"
              className="chakra-button css-1y85x9m"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden={true}
                fill="currentColor"
                focusable={false}
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          className="css-17xejub"
        />
        <div
          className="css-0"
        >
          <h2
            className="chakra-heading css-1c7bccy"
          >
            Dashboard
          </h2>
        </div>
        <div
          className="css-17xejub"
        />
      </div>
    </div>
    <div
      className="css-gmuwbf"
    >
      <div
        className="css-brfpum"
      >
        <div
          className="css-15ddyrp"
          onClick={[Function]}
        >
          <div
            className="css-gmuwbf"
          >
            <div
              className="chakra-stack css-10j85wf"
            >
              <GaugeChart
                colors={
                  Array [
                    "#FF9F46",
                    "#76CE6F",
                  ]
                }
                hideText={true}
                id="gauge-chart"
                nrOfLevels={20}
                percent={0}
              />
              <div
                className="chakra-stack css-jp20jv"
              >
                <p
                  className="chakra-text css-1srjgni"
                >
                  You have completed
                </p>
                <p
                  className="chakra-text css-68kehq"
                >
                  0
                  %
                </p>
                <p
                  className="chakra-text css-1bsg0uq"
                >
                  of this year's work points
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-15ddyrp"
          onClick={[Function]}
        >
          <div
            className="css-gmuwbf"
          >
            <div
              className="chakra-stack css-10j85wf"
            >
              <div
                className="chakra-stack css-84zodg"
              >
                <button
                  aria-label="Calendar"
                  className="chakra-button css-ozg1ic"
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    fill="currentColor"
                    focusable={false}
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 3 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 2.5a1.5 1.5 0 1 0 3 0 1.5 1.5 0 0 0-3 0zm0 5a1.5 1.5 0 1 0 3 0 1.5 1.5 0 0 0-3 0zM1.5 14a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
                      fillRule="evenodd"
                    />
                  </svg>
                </button>
                <h2
                  className="chakra-heading css-1smtq8o"
                >
                  Next Event
                </h2>
              </div>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-svjswr"
              />
              <p
                className="chakra-text css-ykwwr6"
              >
                
              </p>
              <p
                className="chakra-text css-1bsg0uq"
              >
                
              </p>
              <p
                className="chakra-text css-1bsg0uq"
              >
                
              </p>
            </div>
          </div>
        </div>
        <div
          className="css-9nem3t"
        >
          <div
            className="css-gmuwbf"
          >
            <div
              className="chakra-stack css-nscsg6"
            >
              <h2
                className="chakra-heading css-1dklj6k"
              >
                Important Links
              </h2>
              <hr
                aria-orientation="horizontal"
                className="chakra-divider css-svjswr"
              />
              <ul
                className="css-1avoc21"
                role="list"
              >
                <a
                  className="chakra-link css-k6vkpw"
                  href="https://palmyramx.com/pages/track-rules"
                  target="_blank"
                >
                  <li
                    className="css-81gbng"
                  >
                    Member Rules
                  </li>
                </a>
                <a
                  className="chakra-link css-k6vkpw"
                  href="https://twitter.com/palmyramx"
                  target="_blank"
                >
                  <li
                    className="css-81gbng"
                  >
                    Twitter
                  </li>
                </a>
                <a
                  className="chakra-link css-k6vkpw"
                  href="https://www.facebook.com/palmyramx/"
                  target="_blank"
                >
                  <li
                    className="css-81gbng"
                  >
                    Facebook
                  </li>
                </a>
                <a
                  className="chakra-link css-k6vkpw"
                  href="https://www.instagram.com/palmyramx_hogback_hill/?hl=en"
                  target="_blank"
                >
                  <li
                    className="css-81gbng"
                  >
                    Instagram
                  </li>
                </a>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;
